### DISCUSSION TITLE
how to use shell_to_meterpeter

### DISCUSSION THREAD

**Discussion by Salazar33** [rapid7/metasploit-framework#17005] [Q&A] [answered]
can any one please tell me that how can i use shell_to_meterpeter from this framework

**Comment by bcoles** [comment, marked as answer]
`post/multi/manage/shell_to_meterpreter` is a module and can be used like any other module. It is a `post` module so requires setting the `session` ID of the session to be upgraded:

```
use post/multi/manage/shell_to_meterpreter
set session <session id>
```

See also:

* https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/manage/shell_to_meterpreter.md

There are a couple of known issues with this module which have not been patched:

* The module fails on Windows XP SP3 - #16785
* The module fails on systems with non-english localisation. This will be fixed in #16965.
